eng/tools/spec-gen-sdk-runner/src/commands.ts

```diff
@@ -250,6 +250,7 @@ function getExecutionReport(commandInput: SpecGenSdkCmdInput): any {
     commandInput.workingFolder,
     `${commandInput.sdkRepoName}_tmp/execution-report.json`,
   );
+  console.log(`##vso[task.setVariable variable=GeneratedSDK.ExecutionReportPath]${executionReportPath}`);
```

Comment:
Can you explain why the external steps in pipeline need to access `execution-report.json`? Can we output the information directly as pipeline variables?

Reply:
We can but, we need to output a list of service names. Is that possible with run time variables?

Reply:
How about storing all service names as a single delimited string in a pipeline variable, and split it as needed?

Reply:
That could work.

Reply:
Is this only consumed by the one step to improve logging? If so, why not log it from this runner instead? Also isn't the file uploaded as an artifact that could be consumed?

Reply:
It's not for logging. We need some of the properties in the files to use in subsequent steps. And is not published as an artifact.

Reply:
What do we need from the report and who needs it?

Reply:
I talked with Scott about the case to generate package info for multiple package scenario and learned that he met same situation when working on pull request pipeline for SDK language repos. That's why [this approach](https://github.com/Azure/azure-sdk-tools/blob/aba9d53115630fa48f51070d8363da9f1b670309/eng/common/pipelines/templates/steps/save-package-properties.yml#L28) exists. We could try to validate if that will work for `spec-gen-sdk` PR scenario.